src/main/resources/nt_base/html/base.jahiaTemplateScriptFilterRenderingFailure.jsp

```diff
@@ -0,0 +1,40 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="fn" uri="http://java.sun.com/jsp/jstl/functions" %>
+<%@ taglib prefix="functions" uri="http://www.jahia.org/tags/functions" %>
+<%@ taglib prefix="jcr" uri="http://www.jahia.org/tags/jcr" %>
+<%@ taglib prefix="template" uri="http://www.jahia.org/tags/templateLib" %>
+<%@ taglib prefix="utility" uri="http://www.jahia.org/tags/utilityLib" %>
+<%--@elvariable id="currentNode" type="org.jahia.services.content.JCRNodeWrapper"--%>
+<%--@elvariable id="renderContext" type="org.jahia.services.render.RenderContext"--%>
+<%--@elvariable id="error" type="java.lang.Exception"--%>
+<%--@elvariable id="printedError" type="java.lang.String"--%>
+<%--@elvariable id="originalViewKey" type="java.lang.String"--%>
+<%--@elvariable id="originalViewPath" type="java.lang.String"--%>
+
+<c:if test="${renderContext.editMode}">
+    <template:addResources key="jahiaRenderingFullErrorToggler">
+        <script>
+            function jahiaRenderingFullErrorToggleStackTrace(uuid) {
+                let elements = document.querySelectorAll('.jahiaRenderingFullErrorStackTrace_' + uuid);
+                elements.forEach(function(element) {
+                    if (element.style.display === "none") {
+                        element.style.display = "block";
+                    } else {
+                        element.style.display = "none";
+                    }
+                });
+            }
+        </script>
+    </template:addResources>
+
+    <p>
+        <fmt:message key="renderFailure.errorOccurredInView"/>&nbsp;<strong data-sel-role="renderingFailureViewKey">${not empty originalViewKey ? fn:escapeXml(originalViewKey) : ''}</strong>.<br>
+        <fmt:message key="renderFailure.viewPath"/>&nbsp;<strong data-sel-role="renderingFailureViewPath">${not empty originalViewPath ? fn:escapeXml(originalViewPath) : ''}</strong>.<br>
+        <fmt:message key="renderFailure.errorSays"/>&nbsp;<strong data-sel-role="renderingFailureErrorMessage">${not empty error && not empty error.message ? fn:escapeXml(error.message) : ''}</strong>.<br>
+        <button data-sel-role="renderingFailureToggleFullError" onclick="jahiaRenderingFullErrorToggleStackTrace('${currentNode.identifier}')"><fmt:message key="renderFailure.toggleFullError"/></button>
+    </p>
+    <div data-sel-role="renderingFailureFullError" class="jahiaRenderingFullErrorStackTrace_${currentNode.identifier}" style="display: none">
```

Comment:
I think using the node identifier as a class might cause a problem if the same content is twice in the same page, by using a content reference. In that case it will open and close all of the rendered views at the same time which is not what a user would expect. It would also allow to replace the queryAll call with a getElementById which should be more efficient as well. The biggest problem is finding a unique key as it depends on the context.

Reply:
Yep I saw it too, but to me it was not a big deal to display multiple full stack in the page, as long as it was the same node.
But I added a unique errorId now, coming from the backend.